File: Profile/src/main/java/com/example/Profile/model/Comment.java
package com.example.Profile.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.Getter;
import lombok.Setter;

@Setter
@Getter
@Document("comments")
public class Comment {
	@Id
	private Long id;
	private String text;
	private Post post;
	private UserPost userPost;
	private User user;
	
	public Comment(Long id, String text, Post post, User user) {
		super();
		this.id = id;
		this.text = text;
		this.post = post;
		this.user = user;
	}
	
	public Comment(Long id, String text, UserPost userPost, User user) {
		super();
		this.id = id;
		this.text = text;
		this.userPost = userPost;
		this.user = user;
	}

	public Comment() {}
	public Long getId() {
		return id;
	}
	public void setId(Long id) {
		this.id = id;
	}
	public String getText() {
		return text;
	}
	public void setText(String text) {
		this.text = text;
	}
	public Post getPost() {
		return post;
	}
	public void setPost(Post post) {
		this.post = post;
	}
	public UserPost getUserPost() {
		return userPost;
	}
	public void setUserPost(UserPost userPost) {
		this.userPost = userPost;
	}

	public User getUser() {
		return user;
	}

	public void setUser(User user) {
		this.user = user;
	}
	
}

File: client-app/src/app/model/comment.ts
import { UserPost } from './userPost';
import { Post } from "./post";
import { User } from './user';


export interface CommentInterface {
  id?:number;
  text: string;
  post:Post;
  userPost:UserPost;
  user:User;
}
export class Comment implements CommentInterface {
  id?:number;
  text: string;
  post:Post;
  userPost:UserPost;
  user:User;

  constructor(obj: CommentInterface) {
      this.id=obj.id;
      this.text=obj.text;
      this.post=obj.post;
      this.userPost=obj.userPost;
      this.user=obj.user;
  }
}
